### ISSUE TITLE
Stats Panel

### ISSUE THREAD

**Issue by ashfurrow** [johnno1962/injectionforxcode#89]
This would be some panel that shows the user an estimate of the time saved using the plugin. Would use the work from #84 and #88. @orta is going to mock up some design. This is what Text Expander uses, for reference:

![Text Expander is awesome](https://cloud.githubusercontent.com/assets/498212/13676247/05eafb9c-e6b3-11e5-9407-f2f14ddbdab6.png)

**Comment by johnno1962** [comment]
Sounds good! I wonder how you’re going to get the data back from the injectSource.pl script though as only the script knows the path to the iOSInjection project. Perhaps it is best to store the time saved db globally for all projects injected in $HOME/.something.

**Comment by ashfurrow** [comment]
Good point! Hmm. Will need to think about that after some coffee...

**Comment by orta** [comment]
Xcode's user defaults?

On my phone, replies will be curt.

**Comment by johnno1962** [comment]
I’d stick to a couple of dot files in the user’s home directory, one for the total injections and total time saved used by the script and one logging all injections (when, project, build time, injection time) for the stats panel, appended by the script. Xcode’s user defaults are a bit busy/slow for such a db.

**Comment by ashfurrow** [comment]
Makes sense! Is there a common Perl serialization format for the log, or you're thinking just a plaintext log?

**Comment by johnno1962** [comment]
Plain old tab separated?

**Comment by ashfurrow** [comment]
Cool cool 👍

**Comment by orta** [comment]
Alright, so - think about what currently exists:

![screen shot 2016-03-14 at 5 37 08 pm](https://cloud.githubusercontent.com/assets/49038/13760810/6c2beee2-ea0b-11e5-8ee6-e7a266736c8b.png)

and

![screen shot 2016-03-14 at 5 37 16 pm](https://cloud.githubusercontent.com/assets/49038/13760809/6af13c8a-ea0b-11e5-897d-0f4d693e95a4.png)

( and this proposal )

I'd like to talk about splitting the responsibility of some of the user-facing things into two separate places. An Injection Preferences panel, and a Tunable Parameters panel. With the prefs being focused on global state, and the parameters being about the current injection information. I don't have a mental space in this model for the Injection log so far given that it's often shown and is useful, and maybe it should stay as an external window. I do have a useful feature idea for it though

### Injection Wide Settings Panel

- [ ] Inject Storyboards
- [ ] File Watcher
- [ ] File Unlock Command 
- [ ] Change Keyboard Shortcut
- [ ] Silent Mode
- [ ] Stats Overview
- [ ] Currently Connected Devices ( ATM shown on the Xcode badge )

### Tunable Parameters

- [ ] 5 Sliders
- [ ] 5 Colors
- [ ] Remove the Image thing?
- [ ] Last String 

### Injection Console

- [ ] Hide on Successful Build

**Comment by orta** [comment]
I think the Settings panel can be turned into tabs, which can make it easier to focus collections of similar settings too, a good example of this is Alfred, which does it way, way better than I'd expect Injection to

![screen shot 2016-03-14 at 5 39 02 pm](https://cloud.githubusercontent.com/assets/49038/13760998/5f8c40aa-ea0c-11e5-96db-7ced83e96641.png)

![screen shot 2016-03-14 at 5 44 57 pm](https://cloud.githubusercontent.com/assets/49038/13761020/7c7358f2-ea0c-11e5-82f5-755e04b67123.png)


![screen shot 2016-03-14 at 5 39 00 pm](https://cloud.githubusercontent.com/assets/49038/13761001/61b06d3e-ea0c-11e5-8a57-4a2bdcbbb75d.png)

**Comment by johnno1962** [comment]
I’m all ears.. I only have two requirements. A overall desire to keep things as simple as possible and whatever you do must absolutely never crash Xcode! I can’t offer much in the way of input until I see any specific mock-ups you do. Tabs seems a little OTT to me given how few options there are though I agree on the preferences tuning split :+1:

**Comment by johnno1962** [comment]
I’m warming to tabs. what about preferences, tuning, Stats Panel tabs in a single window?

**Comment by orta** [comment]
> I’m warming to tabs. what about preferences, tuning, Stats Panel tabs in a single window?

I mainly wanted to see tuning separated out, as it's about the immediate connection, whereas Pref/Stats are about Injection in general

**Comment by johnno1962** [comment]
fair enough, It’s entirely up to you.

**Comment by johnno1962** [comment]
I’ve created a branch for our furture endeavours “injection2” and given you push access. I hope to convert injection to ARC to make life easier soon

**Comment by orta** [comment]
👍🏾 - not sure how much time I'll have this next 2 weeks to help tout, heading out to siberia, but will be paying attention
